Code/Framework/AzFramework/AzFramework/Network/IRemoteTools.h

```diff
@@ -0,0 +1,262 @@
+            , m_isBlobOwner(false)
+            , m_immediateSelfDispatch(false)
+        {
+        }
+        RemoteToolsMessage(AZ::u64 msgId)
+            : m_msgId(msgId)
+            , m_senderTargetId(0)
+            , m_customBlob(nullptr)
+            , m_customBlobSize(0)
+            , m_refCount(0)
+            , m_isBlobOwner(false)
+            , m_immediateSelfDispatch(false)
+        {
+        }
+
+        virtual ~RemoteToolsMessage();
+
+        //! Add a custom data blob to a Remote Tools message
+        //! @param blob Pointer to the start of the data blob
+        //! @param blobSize Size of the data blob
+        //! @param ownBlob Whether this message owns the life cycle of the blob memory
+        void AddCustomBlob(const void* blob, size_t blobSize, bool ownBlob = false);
+
+        //! Sets if this message can be immediately dispatched if the receipient is itself
+        //! @param immediateSelfDispatchEnabled Whether this message can be immediately dispatched
+        void SetImmediateSelfDispatchEnabled(bool immediateSelfDispatchEnabled);
+
+        //! Gets if this message can be immediately dispatched if the receipient is itself
+        //! @return Whether this message can be immediately dispatched
+        bool IsImmediateSelfDispatchEnabled() const;
+
+        //! Gets the custom data blob assosciated with this message
+        //! @return Memory address of the start of the blob or nullptr if there is none
+        const void* GetCustomBlob() const;
+
+        //! Gets the size of the custom data blob associated with this message
+        //! @return Size of the data blob or 0 if there is none
+        size_t GetCustomBlobSize() const;
+
+        //! Gets the numeric ID of this message
+        //! @return The ID of this message
+        AZ::u64 GetId() const;
+
+        //! Gets this message's sender's ID
+        //! @return ID of the message's sender
+        AZ::u32 GetSenderTargetId() const;
+
+        //! Sets the ID of this message's sender
+        //! @param senderTargetId The ID to set for this message's sender
+        void SetSenderTargetId(AZ::u32 senderTargetId);
+
+    protected:
+        AZ::u64 m_msgId;
+        AZ::u32 m_senderTargetId;
+        const void* m_customBlob;
+        AZ::u32 m_customBlobSize;
+        bool m_isBlobOwner;
+        bool m_immediateSelfDispatch;
+
+        //---------------------------------------------------------------------
+        // refcount
+        //---------------------------------------------------------------------
+        template<class T>
+        friend struct AZStd::IntrusivePtrCountPolicy;
+        AZ_FORCE_INLINE void add_ref()
+        {
+            ++m_refCount;
+        }
+        AZ_FORCE_INLINE void release()
+        {
+            if (--m_refCount == 0)
+            {
+                delete this;
+            }
+        }
+
+        size_t m_refCount;
+        //---------------------------------------------------------------------
+    };
```

Comment:
You can just have RemoteToolsMessage inherit from `AZStd::intrusive_refcount<size_t>` instead https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/smart_ptr/intrusive_refcount.h#L27-L28